test/evm.js

```diff
@@ -1,59 +1,57 @@
-const { ethers } = require("hardhat")
+const {
+  time,
+  mine,
+  takeSnapshot,
+} = require("@nomicfoundation/hardhat-network-helpers")
+const hre = require("hardhat")
+const provider = hre.network.provider
 
-let snapshots = []
+const snapshots = []
 
 async function snapshot() {
-  const id = await ethers.provider.send("evm_snapshot")
+  const snapshot = await takeSnapshot()
   const time = await currentTime()
-  const automine = await ethers.provider.send("hardhat_getAutomine")
-  snapshots.push({ id, time, automine })
+  const automine = await provider.send("hardhat_getAutomine")
+  snapshots.push({ snapshot, automine, time })
 }
 
 async function revert() {
-  const { id, time, automine } = snapshots.pop()
-  await ethers.provider.send("evm_revert", [id])
-  await ethers.provider.send("evm_setNextBlockTimestamp", [time])
-  await ethers.provider.send("evm_setAutomine", [automine])
+  const { snapshot, time, automine } = snapshots.pop()
+  if (snapshot) {
+    await snapshot.restore()
+    await setNextBlockTimestamp(time)
+    await provider.send("evm_setAutomine", [automine])
+  }
 }
 
-/**
- * Enables or disables Hardhat's automine mode.
- *
- * When automine mode is disabled, transactions that revert are silently ignored!
```

Comment:
I'd like to keep this warning around, unless the new hardhat no longer has this behaviour?